### PR TITLE
Add min interval prometheus metric

### PR DESCRIPTION
**Pull request by MaksTkachuk**
This is needed to be able to know which metric has which minimal interval, to then build and monitor service level indicators for latency of metrics

### CODE REVIEW

File: api_tests/db/migrations.py [hunk 1]
```diff
@@ -13,6 +13,9 @@ def run_migrations():
     if not _exist_in_table('response', 'gqltestcase'):
         migrate(migrator.add_column('gqltestcase', 'response', TextField(default='')))
 
+    if not _exist_in_table('min_interval', 'gqltestcase'):
+        migrate(migrator.add_column('gqltestcase', 'min_interval', TextField(default='')))
```
[the remark this answers is not in the document]
Comment (MaksTkachuk) [reply api_tests/db/migrations.py:17]: added not-null constraint
min_interval will be a label so then we can group metrics by that label

File: app.py [hunk 2]
```diff
@@ -241,13 +241,19 @@ def generate_prometheus_metrics(test_suite):
...
+    lines_min_interval = [
+        '# HELP api_tests_min_interval Minimal interval for a metric',
+        '# TYPE api_tests_min_interval gauge'
+    ]
     for project in data:
         for item in project['data']:
             line_status = f'api_tests_status{{asset="{project["slug"]}", metric="{item["name"]}", status="{item["status"]}"}} 1'
             line_delay = f'api_tests_last_datapoint_timestamp{{asset="{project["slug"]}", metric="{item["name"]}"}} {item["last_datapoint_timestamp"]}'
             line_performance = f'api_tests_query_duration{{asset="{project["slug"]}", metric="{item["name"]}"}} {item["elapsed_time"]}'
+            line_min_interval = f'api_tests_min_interval{{asset="{project["slug"]}", metric="{item["name"]}", min_interval="{item["min_interval"]}"}} 1'
```
[the remark this answers is not in the document]
Comment (MaksTkachuk) [reply app.py:253]: this is a dummy value as the meaning of this metric is in the label

File: app.py [hunk 3]
```diff
@@ -241,13 +241,19 @@ def generate_prometheus_metrics(test_suite):
         '# HELP api_tests_query_duration Time taken to run the query in seconds',
         '# TYPE api_tests_query_duration gauge'
     ]
+    lines_min_interval = [
+        '# HELP api_tests_min_interval Minimal interval for a metric',
+        '# TYPE api_tests_min_interval gauge'
```
[the remark this answers is not in the document]
Comment (MaksTkachuk) [reply app.py:246]: value will always be 1

File: app.py [hunk 4]
```diff
@@ -241,13 +241,19 @@ def generate_prometheus_metrics(test_suite):
...
             line_delay = f'api_tests_last_datapoint_timestamp{{asset="{project["slug"]}", metric="{item["name"]}"}} {item["last_datapoint_timestamp"]}'
             line_performance = f'api_tests_query_duration{{asset="{project["slug"]}", metric="{item["name"]}"}} {item["elapsed_time"]}'
+            line_min_interval = f'api_tests_min_interval{{asset="{project["slug"]}", metric="{item["name"]}", min_interval="{item["min_interval"]}"}} 1'
             lines_status.append(line_status)
             lines_delay.append(line_delay)
             lines_performance.append(line_performance)
+            lines_min_interval.append(line_min_interval)
 
-    return '\n'.join(sorted(lines_status)) + '\n' + '\n'.join(sorted(lines_delay)) + '\n' + '\n'.join(sorted(lines_performance))
+    return '\n'.join(sorted(lines_status)) + '\n' + '\n'.join(sorted(lines_delay)) + '\n' + '\n'.join(sorted(lines_performance)) + '\n' + '\n'.join(sorted(lines_min_interval))
```
[the remark this answers is not in the document]
Comment (MaksTkachuk) [reply app.py]: fixed

**Review by kalkov — APPROVED** [review]

**Pull request merged**